app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -253,9 +265,20 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
 
         components.core.requestInterceptor.setNavigationController(navHost.navController)
 
+        waitForNavGraphInflation(navHost.navController)
+
         StartupTimeline.onActivityCreateEndHome(this) // DO NOT MOVE ANYTHING BELOW HERE.
     }
 
+    /**
+     * This job inflates the NavGraph and attaches it to our NavController.
+     */
+    private fun createNavGraphAsync() = MainScope().launch(Dispatchers.Default) {
```

Comment:
Why is this on `MainScope()`? I don't think it adds anything vs. `GlobalScope.launch`. Is it so we can cancel the job? In that case, I think we need to manually add code to cancel `MainScope()`, according to the docs: https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-main-scope.html